pkg/certificates/bootstrap/cert-manager.go

```diff
@@ -220,3 +228,66 @@ func (f *FileCertificateManager) loadCertificates() (serverCrt *tls.Certificate,
 	crt.Leaf = leaf[0]
 	return &crt, nil
 }
+
+type SecretCertificateManager struct {
+	store     cache.Store
+	secretKey string
+	tlsCrt    string
+	tlsKey    string
+	crtLock   *sync.Mutex
+	revision  string
+	crt       *tls.Certificate
+}
+
+func (s *SecretCertificateManager) Start() {
+}
+
+func (s *SecretCertificateManager) Stop() {
+}
+
+func (s *SecretCertificateManager) Current() *tls.Certificate {
+	s.crtLock.Lock()
+	s.crtLock.Unlock()
```

Comment:
Is the defer missing?
```suggestion
	defer s.crtLock.Unlock()
```

Reply:
uh right.

Reply:
Fixed.